src/utils/relativeId.ts

```diff
@@ -11,7 +10,7 @@ export default function relativeId(id: string) {
 	return relative(resolve(), id);
 }
 
-export function isPlainPathFragment(name: string) {
+export function isPlainPathFragment(name: string, sanitizeFileName: (fileName: string) => string) {
```

Comment:
I am still not 100% happy we are sanitising in this function as this will throw a sanitation error even if the user opted out of the feature. As I said, I would remove the parameter here and from `hasValidName`, and maybe adjust the error message when `hasValidName` fails as sanitising will happen anyway later when `renderNamePatter` is invoked.
What do you think?

Reply:
Admittedly, it is possible that `isAbsolute` will then throw, but I would see this as a minor issue, and it will likely not throw in the browser build where we use a simple polyfill.

Reply:
I agree separation would be better here, will try and implement that.